Gems/RemoteTools/Code/Source/RemoteToolsSystemComponent.cpp

```diff
@@ -363,17 +361,26 @@ namespace RemoteTools
         AzNetworking::INetworkInterface* networkInterface =
             AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(
                 AZ::Name(m_entryRegistry[target.GetPersistentId()].m_name));
-        
-        OutboundToolingDatum datum;
-        datum.first = target.GetPersistentId();
-        datum.second.swap(msgBuffer);
-        m_outboxThread->PushOutboxMessage(
-            networkInterface, static_cast<AzNetworking::ConnectionId>(target.GetNetworkId()), AZStd::move(datum));
-        if (!m_outboxThread->IsRunning())
+
+        auto connectionId = static_cast<AzNetworking::ConnectionId>(target.GetNetworkId());
+        const uint8_t* outBuffer = reinterpret_cast<const uint8_t*>(msgBuffer.data());
+        const size_t totalSize = msgBuffer.size();
+        size_t outSize = totalSize;
+        while (outSize > 0 && networkInterface != nullptr)
         {
-            m_outboxThread->Join();
-            m_outboxThread->Start();
+            // Fragment the message into RemoteToolsMessageBuffer packet sized chunks and send
+            size_t bufferSize = AZStd::min(outSize, aznumeric_cast<size_t>(RemoteToolsBufferSize));
+            RemoteToolsPackets::RemoteToolsMessage tmPacket;
+            tmPacket.SetPersistentId(target.GetPersistentId());
+            tmPacket.SetSize(aznumeric_cast<uint32_t>(totalSize));
```

Comment:
Just checking, the size of the `tmPacket` in every loop is always the full size `totalSize` and not the size of the current message buffer `bufferSize`?

Reply:
Even though the message may get broken into several pieces of maximum `16,000` bytes, the tmPacket.SetSize() must be the TotalSize. In other words, all of the submessages carry the TotalSize with them, in addition the message owns a subpacket encoded as RemoteToolsMessageBuffer, which has the subpacket size, in this case `bufferSize`.

Long story short, ` tmPacket.SetSize(aznumeric_cast<uint32_t>(totalSize));` is correct.

Reply:
Thanks for the clarification.